### DISCUSSION TITLE
Pressing `esc` to cancel `find`  when it is not visible is non-intuitive.

### DISCUSSION THREAD

**Discussion by amano-kenji** [sxyazi/yazi#1768] [1. Q&A]
yazi is mostly intutive, but cancelling invisible `find` by pressing `esc` was non-intuitive.

I had to retrace my steps to even figure out that invisible `find` in the parent directory was cancelled before `esc` can cancel a selection in the current directory.

Until I figured out that I cancelled invisible `find`, I thought I triggered a bug in yazi.

I think yazi should improve intuitiveness in this case. Perhaps, `find` can be more visible anywhere, or `find` is cancelled when leaving or entering a directory by default?

**Comment by amano-kenji** [comment]
One possible solution is to escape find only when it is visible or active in the current directory.

**Comment by sxyazi** [comment]
Close in favor of https://github.com/sxyazi/yazi/issues/1825
